### PR TITLE
Dev/1.15.2

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java [hunk 1]
```diff
@@ -284,6 +326,328 @@ public static String httpGet(String url, String header, String body, String outp
...
+    return httpRequest(url, header, body, HTTPX_DELETE_STRING, outputFile);
+  }
+
+  /**
+   * Performs an HTTP OPTIONS request to the specified URL using the specified request header and
+   * body.
+   *
+   * @param url URL to make request
+   * @param header request header
+   * @param body request body
```
Comment (it-hms) [inline src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java:496]: options request does not have a body. I think it makes the most sense to remove parameter

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java [hunk 2]
```diff
@@ -284,6 +326,328 @@ public static String httpGet(String url, String header, String body, String outp
...
+  }
+
+  /**
+   * Performs an HTTP OPTIONS request to the specified URL using the specified request header and
+   * body, then outputs the response to the specified file. If the specified file already exists, it
+   * will be overwritten.
+   *
+   * @param url URL to make request
+   * @param header request header
+   * @param body request body
```
Comment (it-hms) [inline src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java:523]: options request does not have a body

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java [hunk 3]
```diff
@@ -284,6 +326,328 @@ public static String httpGet(String url, String header, String body, String outp
...
+          SCHttpUnknownException {
+    return httpRequest(url, header, body, HTTPX_OPTIONS_STRING, outputFile);
+  }
+
+  /**
+   * Performs an HTTP HEAD request to the specified URL using the specified request header and body.
+   *
+   * @param url URL to make request
+   * @param header request header
+   * @param body request body
```
Comment (it-hms) [inline src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java:549]: HEAD request does not have a body

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java [hunk 4]
```diff
@@ -284,6 +326,328 @@ public static String httpGet(String url, String header, String body, String outp
...
+  }
+
+  /**
+   * Performs an HTTP HEAD request to the specified URL using the specified request header and body,
+   * then outputs the response to the specified file. If the specified file already exists, it will
+   * be overwritten.
+   *
+   * @param url URL to make request
+   * @param header request header
+   * @param body request body
```
Comment (it-hms) [inline src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java:576]: nobody

**Review by it-hms — CHANGES_REQUESTED** [review]
I think  my only request was to remove body for methods that do not support a body (according to the RFCs)

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java (hunk 4 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java]: Removed (replaced with "")

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java (hunk 3 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java]: Removed (replaced with "")

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java (hunk 2 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java]: Removed (replaced with "")

File: src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java (hunk 1 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java]: Removed (replaced with "")

**Review by TomKimsey — APPROVED** [review]

**Review by it-hms — APPROVED** [review]

**Pull request merged**